sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceRDD.scala

```diff
@@ -83,7 +84,7 @@ private class PartitionIterator[T](reader: PartitionReader[T]) extends Iterator[
 
   override def next(): T = {
     if (!hasNext) {
-      throw new java.util.NoSuchElementException("End of stream")
+      throw QueryExecutionErrors.endOfStreamError()
```

Comment:
`NoSuchElementException` is generated during execute and should be placed in `QueryExecutionErrors`.

Reply:
I believe this is already located in `QueryExecutionErrors`.

Reply:
> I believe this is already located in `QueryExecutionErrors`.

Oh. I see. But fix the other comments please.